### PR TITLE
soc: mcxw: Fix boot and enable timers

### PR DESCRIPTION
**Pull request by decsny**
- Enable some of the timers on MCXW71 (TPM, LPTMR, WDOG : PWM, counter, watchdog)

- Fix boot issues caused by changes to upstream platform init while waiting for merge of #77434

- Some other minor adjustments

### CODE REVIEW

**Review by henrikbrixandersen — DISMISSED** [review]

**Review by danieldegrasse — COMMENTED** [review]
YAML and doc pages for this board need to be updated to indicate support for counter and watchdog. Otherwise, PR LGTM

File: soc/nxp/mcx/mcxw/mcxw71_platform_init.S [hunk 1]
```diff
@@ -16,13 +16,11 @@
...
-#ifdef CONFIG_PLATFORM_SPECIFIC_INIT
 
+GTEXT(soc_reset_hook)
+SECTION_SUBSEC_FUNC(TEXT,_reset_section,soc_reset_hook)
 
-GTEXT(z_arm_platform_init)
-SECTION_SUBSEC_FUNC(TEXT,_reset_section,z_arm_platform_init)
-
-.z_arm_platform_init:
+.soc_reset_hook:
```
Comment (axelnxp) [inline soc/nxp/mcx/mcxw/mcxw71_platform_init.S:23]: I would suggest to have a dedicated function for the ECC ram init, and have `soc_reset_hook` calling it.
The reason behind is we may need to call the ECC ram init as a standalone procedure, outside the early init.
For instance, at wake up from power down mode, if an ECC ram bank has been shutdown, we need to go through this procedure again, so we could call it directly.

This is why in #79060, we implemented `soc_reset_hook` in C, and the ECC init function separately.
Having `soc_reset_hook` in C would likely be "easier" to maintain and extend if needed.

Comment (axelnxp) [reply soc/nxp/mcx/mcxw/mcxw71_platform_init.S:23]: Where is placed the stack ? STCM3 and STCM4 are non-ecc (range 0x30010000 - 0x30019FFF)

Comment (decsny) [reply soc/nxp/mcx/mcxw/mcxw71_platform_init.S:23]: and also, to answer your question, the stack can be anywhere, it just goes at the end of the image. Whether or not it goes in non-ecc range is just depending on the size of the image.

**Review by EmilioCBen — APPROVED** [review]

**Review by danieldegrasse — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Review by mmahadevan108 — APPROVED** [review]

**Pull request merged**